client/checkout/woopay/email-input-iframe.js

```diff
@@ -479,6 +489,7 @@ export const handleWooPayEmailInput = async (
 
 		clearTimeout( timer );
 		spinner.remove();
+		deleteSkipWooPayCookie();
```

Comment:
I love that you caught the case I didn't where the WooPay flow can be started through the email input and not just through the button 👏 ❤️ 

That said, I'm not so sure removing the cookie on just any input to the email field is the best way to go about this. Did you have any specific reason for removing the cookie on every input to the field as opposed to when a valid WooPay email is entered or right before the redirect to WooPay?

***

If we clear it only when a valid WooPay email is entered into the field I think we can do that in the `openIframe` function, since the iframe _should_ only be opened when the WooPay flow has been triggered with a valid email:

https://github.com/Automattic/woocommerce-payments/blob/6b7b29b9666c911e098cb1ef84f8ec5911e0c08e/client/checkout/woopay/email-input-iframe.js#L421-L423

***

If we clear it right before the redirect I think we could do something like:

```js
const redirectToWooPay = ( wooPayUrl ) => {
    deleteSkipWooPayCookie();
    window.location = wooPayUrl;
};

// And then call this function everywhere we redirect to WooPay in the 'message' event handler.
```

Reply:
Good call out, thanks for thinking this through! Performing the deletion in the `input` event is not ideal indeed. I've changed this in https://github.com/Automattic/woocommerce-payments/pull/8570/commits/45a82b00eac8fcd1b6c7b725204095bed000f88c.

I like the first alternative you proposed but I went with the second. I didn't choose the first because I thought about the scenario when the shopper could dismiss the modal and don't proceed to the WooPay checkout. In that case, I don't think it makes sense to delete the cookie. So, to get this behavior consistent and match the WooPay express checkout button approach, which will primarily go through the first party auth and straight redirect to WooPay, I went with the second option

Reply:
Yeah, I also realized that I don't think the OTP window may not be shown every time, so definitely the right call. Thank you César!